be/src/column/chunk.cpp

```diff
@@ -206,10 +210,18 @@ Status Chunk::deserialize(const uint8_t* src, size_t len, const RuntimeChunkMeta
         src = column->deserialize_column(src);
     }
 
-    size_t except = serialize_size();
-    if (UNLIKELY(len != except)) {
-        return Status::InternalError(
-                strings::Substitute("deserialize chunk data failed. len: $0, except: $1", len, except));
+    // The logic is a bit confusing here.
+    // `len` and `expected` are both "estimated" serialized size. it could be larger than real serialized size.
+    // `serialized_size` and `read_size` are both "real" serialized size. it's exactly how much bytes are written into buffer.
+    // For some object column types like bitmap/hll/percentile, "estimated" and "real" are not always the same.
+    // And fr bitmap, sometimes `len` and `expected` are different. So to fix that problem, we fallback to compare "real" serialized size.
+    size_t read_size = src - head;
+    size_t expected = serialize_size();
+
+    if (UNLIKELY(len != expected && read_size != serialized_size)) {
```

Comment:
could check `read_size != serialized_size` firstly, then in most case, we need't invoke `serialize_size();`

Reply:
That's a great idea.

For older version of BE, there is no `serialized_size`, we fallback to compare `len` and `expected_size`.

And For new version of BE, we compare `read_size` and `serialized_size` and have no need to call `serialize_size()`.